pontoon/actionlog/migrations/0004_auto_20240816_1045.py

```diff
@@ -0,0 +1,81 @@
+# Generated by Django 4.2.11 on 2024-08-16 10:45
+
+import datetime
+
+from django.db import migrations
+
+
+def migrate_translation_to_actionlog(apps, schema_editor):
+    Translation = apps.get_model("base", "Translation")
+    ActionLog = apps.get_model("actionlog", "ActionLog")
+
+    translation_info = {
+        "date": (("translation:created", "Translation created"), "user"),
+        "approved_date": (
+            ("translation:approved", "Translation approved"),
+            "approved_user",
+        ),
+        "unapproved_date": (
+            ("translation:unapproved", "Translation unapproved"),
+            "unapproved_user",
+        ),
+        "rejected_date": (
+            ("translation:rejected", "Translation rejected"),
+            "rejected_user",
+        ),
+        "unrejected_date": (
+            ("translation:unrejected", "Translation unrejected"),
+            "unrejected_user",
+        ),
+    }
+
+    # date to end migration
+    end_date = datetime.datetime(
+        2020, 1, 7, 9, 25, 11, 829125, tzinfo=datetime.timezone.utc
+    )
+
+    actions_to_log = []
+
+    # Exclude translations that don't have any user set to improve performance
+    translations = Translation.objects.filter(date__lt=end_date).exclude(
+        user=None,
+        approved_user=None,
+        unapproved_user=None,
+        rejected_user=None,
+        unrejected_user=None,
+    )
```

Comment:
This seems to improve performance by at least 90%:

```suggestion
    translations = (
        Translation.objects.filter(date__lt=end_date)
        .exclude(
            user=None,
            approved_user=None,
            unapproved_user=None,
            rejected_user=None,
            unrejected_user=None,
        )
        .values(
            "id",
            "date",
            "approved_date",
            "unapproved_date",
            "rejected_date",
            "unrejected_date",
            "user",
            "approved_user",
            "unapproved_user",
            "rejected_user",
            "unrejected_user",
        )
    )
```

Reply:
Note that you'll need to make adjustments to the code below to make it work.

Reply:
aight :handshake:  on it

Reply:
Sorry, I already went ahead and pushed the changes.

Reply:
I deployed to our stage server and the migration completed fine! 👍

I'll just check now if the app somewhere relies on the ActionLog entries to be created in chronological order. If not, we can just merge this.

Reply:
okay, if you need me just let me know :) :+1: